tests/garage/tf/algos/test_td3.py

```diff
@@ -0,0 +1,74 @@
+from garage.tf.experiment import LocalTFRunner
+from garage.tf.policies import ContinuousMLPPolicy
+from garage.tf.q_functions import ContinuousMLPQFunction
+from tests.fixtures import snapshot_config, TfGraphTestCase
+
+
+class TestTD3(TfGraphTestCase):
+    """Tests for TD3 algo."""
+
+    def test_td3_pendulum(self):
+        """Test TD3 with Pendulum environment."""
+        with LocalTFRunner(snapshot_config) as runner:
+            env = TfEnv(gym.make('InvertedDoublePendulum-v2'))
+
+            action_noise = GaussianStrategy(env.spec,
+                                            max_sigma=0.1,
+                                            min_sigma=0.1)
+
+            policy = ContinuousMLPPolicy(env_spec=env.spec,
+                                         hidden_sizes=[400, 300],
+                                         hidden_nonlinearity=tf.nn.relu,
+                                         output_nonlinearity=tf.nn.tanh)
+
+            qf = ContinuousMLPQFunction(name='ContinuousMLPQFunction',
+                                        env_spec=env.spec,
+                                        hidden_sizes=[400, 300],
+                                        action_merge_layer=0,
+                                        hidden_nonlinearity=tf.nn.relu)
+
+            qf2 = ContinuousMLPQFunction(name='ContinuousMLPQFunction2',
+                                         env_spec=env.spec,
+                                         hidden_sizes=[400, 300],
+                                         action_merge_layer=0,
+                                         hidden_nonlinearity=tf.nn.relu)
+
+            replay_buffer = SimpleReplayBuffer(env_spec=env.spec,
+                                               size_in_transitions=int(1e6),
+                                               time_horizon=250)
+
+            algo = TD3(env_spec=env.spec,
+                       policy=policy,
+                       policy_lr=1e-3,
+                       qf_lr=1e-3,
+                       qf=qf,
+                       qf2=qf2,
+                       replay_buffer=replay_buffer,
+                       target_update_tau=0.005,
+                       n_epoch_cycles=20,
+                       n_train_steps=1,
+                       discount=0.99,
+                       smooth_return=False,
+                       min_buffer_size=int(1e4),
+                       buffer_batch_size=100,
+                       policy_weight_decay=0.001,
+                       qf_weight_decay=0.001,
+                       exploration_strategy=action_noise,
+                       policy_optimizer=tf.train.AdamOptimizer,
+                       qf_optimizer=tf.train.AdamOptimizer)
+
+            runner.setup(algo, env)
+            last_avg_ret = runner.train(n_epochs=10,
+                                        n_epoch_cycles=20,
+                                        batch_size=250,
+                                        store_algo=False)
```

Comment:
Do you mean `store_paths`?

Reply:
I added `store_algo` to train function because storing TD3 algo was giving some picking error.

Reply:
Hmm, would you explain to me why this picking error happens?

Reply:
This is the exact error I was getting `AttributeError: Can't pickle local object 'compile_function.<locals>.run'`. You can recreate this with following code
```
import pickle

def test():
    def run():
        return "Hello"
    return run

class A:
  def __init__(self):
    self._test = test()

a = A()
pickle.dump(a, open('test.pickle', 'wb'))
```

Reply:
I think you can remove the member function in __getstate__() and restore it in __setstate()__ to make it pickleable, but that may requires more code refactoring.

Reply:
@ahtsan Similar to what is done in `DDPG` ?

Reply:
It would be good to remove this, so that we test that TD3 can be pickled.